src/PrivateKeyLoader.ts

```diff
@@ -68,86 +49,83 @@ export default class PrivateKeyLoader {
-                new KeyknoxManager(
-                    this.toolbox.jwtProvider,
-                    privateKey,
-                    publicKey,
-                    undefined,
-                    new KeyknoxCrypto(this.toolbox.virgilCrypto),
-                ),
-            ),
-            new KeyEntryStorage('keyknox-storage'),
-        );
-        try {
-            await storage.sync();
-        } catch (e) {
-            throw new WrongKeyknoxPasswordError();
-        }
-
-        return storage;
+    hasPrivateKey() {
+        return this.localStorage.exists(this.identity);
     }
 
-    private async initStorage(password: string) {
-        if (!this.syncStorage) this.syncStorage = this.createSyncStorage(password);
-        try {
-            await this.syncStorage;
-        } catch (e) {
-            this.syncStorage = undefined;
-            throw e;
+    private handleResetError = (e: Error) => {
+        if (e instanceof CloudEntryDoesntExistError) {
+            throw new PrivateKeyNoBackupError();
         }
-        return this.syncStorage;
-    }
+        throw e;
+    };
 
-    private generateBrainPair = (password: string) =>
-        this.brainKey.generateKeyPair(password).catch(e => {
-            if (typeof e === 'object' && e.code === 60007) {
+    private async generateBrainPair(pwd: string) {
+        const brainKey = createBrainKey({
+            virgilCrypto: this.toolbox.virgilCrypto,
+            virgilPythiaCrypto: this.pythiaCrypto,
+            accessTokenProvider: this.toolbox.jwtProvider,
+        });
+
+        return await brainKey.generateKeyPair(pwd).catch((e: Error & { code?: number }) => {
+            if (typeof e === 'object' && e.code === BRAIN_KEY_THROTTLING_ERROR_CODE) {
                 const promise = new Promise((resolve, reject) => {
                     const repeat = () =>
-                        this.brainKey
-                            .generateKeyPair(password)
+                        brainKey
+                            .generateKeyPair(pwd)
                             .then(resolve)
                             .catch(reject);
-                    setTimeout(repeat, 2000);
+                    setTimeout(repeat, BRAIN_KEY_RATE_LIMIT_DELAY);
                 });
                 return promise as Promise<KeyPair>;
             }
             throw e;
         });
+    }
+
+    private async getStorage(pwd: string) {
+        const keyPair = await this.generateBrainPair(pwd);
+
+        const storage = new CloudKeyStorage(
+            new KeyknoxManager(
+                this.toolbox.jwtProvider,
+                keyPair.privateKey,
+                keyPair.publicKey,
+                undefined,
+                new KeyknoxCrypto(this.toolbox.virgilCrypto),
+            ),
+        );
+        try {
+            await storage.retrieveCloudEntries();
+        } catch (e) {
+            if (e instanceof VirgilCryptoError) throw new WrongKeyknoxPasswordError();
```

Comment:
please check that `e.name === 'VirgilCryptoError'` instead. Currently `VirgilCryptoError` is an internal type.